basics_of_sst-core/tutorial/Example05.md

```diff
@@ -0,0 +1,651 @@
+            // are initialized as {"lib1.event1", "lib1.event2", "lib2.event3"}.
+            // Note these are the names as strings, not types.
+            //
+            SST_ELI_DOCUMENT_PORTS(
+                { "link_", "Message port", {}}
+            )
+
+            // Register the subcomponent API.
+            //
+            SST_ELI_REGISTER_SUBCOMPONENT_API(Example05::ExampleSubComponent)
+
+            // Register the subcomponent.
+            //
+            SST_ELI_REGISTER_SUBCOMPONENT_DERIVED(
+                ExampleSubComponent,                // Name of the class being registered.
+                                                    // INSERT_CLASS_NAME
+                "example",                          // Element library.  Name of the .so.
+                "ExampleSubComponent",              // Name used to reference the subcomponent.
+                                                    // Can be anything but typically set to
+                                                    // the class name.
+                SST_ELI_ELEMENT_VERSION(1, 0, 0),   // 
+                "Example subcomponent",             // Brief subcomponent description
+                Example05::ExampleSubComponent      // Name of the subcomponent interface the
+                                                    // subcomponent inherits from.  Should be
+                                                    // the full parent class name.
+                                                    // "INSERT_FULL_PARENT_CLASS_NAME" or 
+                                                    // "INSERT_COMPLETE_NAMESPACE::INSERT_PARENT_CLASS_NAME"
+            )
+
+        private:
+            // Links to connect to other component.
+            // The link connects to the subcomponents port.
+            //
+            SST::Link* link_ = nullptr;
+
+            // Count sent to the connected subcomponent.
+            //
+            SST::Output logger_;        // Used for logging messages.
+            uint64_t componentId_;      // Id associated with the parent component.
+            uint64_t slotNumber_;       // Slot number associated with the subcomponent.
+            uint32_t count_ = 0;
+            uint64_t clockTickCount_ = 0;
+    };
+}
+
+#endif
+
+```
+
+The constructors and destructor remain unchanged.  However, there is a new `start()` method used to initialize the subcomponent.  This is where parameters for the subcomponent are processed and the link is initialized.
+
+You may wonder, why not do this initialization in the subcomponent constructor?  For some reason the subcomponent constructors were not being called when created in the parent component.  Use of the `start()` method is a reasonable alternative that models the method used for top level components.  The difference is, this `start()` method will not be called automatically.  It must be called explicitly when the subcomponent is created.
```

Comment:
The constructor is getting called, I put a print() statement inside it in Example05 and see it getting printed. This statement is incorrect.